typing/typing_recovery.mli

```diff
@@ -0,0 +1,51 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                Ulysse Gérard, Thomas Refis, Frederic Bour              *)
+(*                           Xavier Van de Woestyne                       *)
+(*                                                                        *)
+(*   Copyright 2025 Tarides                                               *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+exception Resume
+
+(** Returns a reference initially set to false that will be set to true
+    when a type error is raised. *)
+val monitor_errors : unit -> bool ref
+
+(** Raise an error that can be caught: normal flow is resumed if a
+    [catch_errors] handler was installed. *)
+val raise_error : exn -> unit
+
+(** Any [raise_error] invoked inside catch_errors will be added to
+    the list. *)
+val catch_errors : Warnings.state -> exn list ref -> (unit -> 'a) -> 'a
+
+(** Re-raise an exception*)
+val resume_raise : exn -> 'a
```

Comment:
The docstring is incorrect.
I went and had a look in merlin, which states:
```ocaml

(** Resume after error: like [raise_error], but if a handler was provided a
    Resume exception is raised.  This allows to specify a special case when an
    error is caught. *)
exception Resume

val resume_raise : exn -> 'a
```

Also, `resume_raise` is only used in a single place in typecore, and I feel like it would be cleaner to special case that occurrence and not bother with `resume_raise`.